tree.go

```diff
@@ -351,9 +379,10 @@ func (n *InternalNode) Delete(key []byte) error {
 	case *HashedNode:
 		return errors.New("trying to delete from a hashed subtree")
 	case *LeafNode:
-		if !bytes.Equal(child.key, key) {
+		if !bytes.Equal(child.key[:31], key[:31]) {
 			return errDeleteNonExistent
 		}
+		// overwrite empty if the last level node exists
 		n.children[nChild] = Empty{}
```

Comment:
I feel like you need to call `LeafNode.Delete` here and only replace with `Empty` when all of its values are nil